fossilize_external_replayer_linux.hpp

```diff
@@ -49,10 +49,59 @@
 #include "platform/gcc_clang_spinlock.hpp"
 #endif
 
+#ifdef __linux__
+
 #ifndef SCHED_BATCH
 #define SCHED_BATCH 3
 #endif
 
+/**
+ * Define the syscall interface in Linux because it is missing from glibc
+ * https://www.kernel.org/doc/html/latest/block/ioprio.html
+ */
+
+#ifndef IOPRIO_CLASS_SHIFT
+#define IOPRIO_CLASS_SHIFT (13)
+#endif
+
+#ifndef IOPRIO_PRIO_MASK
+#define IOPRIO_PRIO_MASK ((1UL << IOPRIO_CLASS_SHIFT) - 1)
+#endif
+
+#ifndef IOPRIO_PRIO_CLASS
+#define IOPRIO_PRIO_CLASS(mask) ((mask) >> IOPRIO_CLASS_SHIFT)
+#endif
+
+#ifndef IOPRIO_PRIO_DATA
+#define IOPRIO_PRIO_DATA(mask) ((mask) & IOPRIO_PRIO_MASK)
+#endif
+
+#ifndef IOPRIO_PRIO_VALUE
+#define IOPRIO_PRIO_VALUE(clazz, data) (((clazz) << IOPRIO_CLASS_SHIFT) | data)
+#endif
+
+enum
+{
+	IOPRIO_CLASS_NONE,
+	IOPRIO_CLASS_RT,
+	IOPRIO_CLASS_BE,
+	IOPRIO_CLASS_IDLE,
+};
+
+enum
+{
+	IOPRIO_WHO_PROCESS = 1,
+	IOPRIO_WHO_PGRP,
+	IOPRIO_WHO_USER,
+};
+
+static inline int ioprio_set(int which, int who, int ioprio)
+{
+	return (int)syscall(SYS_ioprio_set, which, who, ioprio);
```

Comment:
Where is SYS_ioprio_set defined? If ioprio defines don't exist, doesn't that mean SYS_ioprio_set might not be defined? I don't see `sys/syscall.h` include at least.

Reply:
Actually, I'm not sure. It seems to just exist... Probably in glibc? If we'd define that ourselves, we would need to do that per platform and word size (32 bit / 64 bit). But research shows it should be in `sys/syscall.h`. BTW, it's there (bits/syscall.h included from sys/syscall.h) for me:
```c
#ifdef __NR_ioprio_get
# define SYS_ioprio_get __NR_ioprio_get
#endif
```

Now the question remains where `__NR_ioprio_get` comes from... Probably comes from `asm/unistd.h` which has a platform-dependent include.